### PR TITLE
make psgo thread-safe

### PR DESCRIPTION
**Pull request by vrothberg**
This PR refactors some global variables into a `type psContext` that's passed around to all descriptor functions.

Fixes: https://github.com/containers/psgo/issues/28

### CODE REVIEW

**Comment by vrothberg** [comment]
Wow. I started writing the PR on a pretty old state without noticing. Need some rewriting to make this work again...

**Comment by vrothberg** [comment]
@rhatdan PTAL. The train has an unkown delay to Paris, so I might find time for more cleanups :^)

File: psgo.go [hunk 1]
```diff
@@ -282,11 +283,13 @@ func JoinNamespaceAndProcessInfo(pid string, descriptors []string) ([][]string,
...
 	}
 
+	ctx := new(psContext)
+
 	// extract data from host processes only on-demand / when at least one
 	// of the specified descriptors requires host data
 	for _, d := range aixDescriptors {
 		if d.onHost {
-			setHostProcesses(pid)
+			ctx.hostProcesses, _ = hostProcesses(pid)
```
Comment (rhatdan) [inline psgo.go]: You don't care if this fails?

**Comment by rhatdan** [comment]
Overall LGTM, just a bit concerned about ignoring errors.

File: psgo.go (hunk 1 above)
Comment (vrothberg) [reply psgo.go]: It has been ignored prior to this change. Now, we return the errors.

Comment (vrothberg) [reply psgo.go]: ... I mean: I updated the code and now we return the errors.

**Comment by vrothberg** [comment]
@rhatdan PTAL. The errors are now returned.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**